### PR TITLE
Automatically setup variables for gpu.renderstages.

### PR DESCRIPTION
**Pull request by lpy**
Right before application activity is started, setup the variables for
gpu.renderstages so that the Vulkan layer is loaded properly.

BUG: b/142082436
Test: manual test with vulkan app

### CODE REVIEW

**Comment by lpy** [comment]
ptal

**Comment by lpy** [comment]
Could you please explain a little bit more about how it should look like? I am using the function: `android.SetupLayer` from `core.os.android`: https://github.com/google/gapid/pull/3258/files#diff-e69c7602ac1d09c87b6580cff082ec50R80

**Comment by pmuetschard** [comment]
Oops, ok. I just saw the constants and assumed you were re-implementing it... Please remove the unused constants.

**Comment by lpy** [comment]
Thanks, I removed unused constants! :)

File: gapis/perfetto/android/trace.go [hunk 1]
```diff
@@ -54,25 +90,28 @@ func Start(ctx context.Context, d adb.Device, a *android.ActivityAction, opts *s
 		}.Bind(ctx)
 	}
 
+	// Before we start the activity, attempt to setup environment if gpu.renderstages is selected.
+	cleanupFunc, err := setupRenderStagesEnvironment(ctx, d, a.Package.Name, opts.PerfettoConfig)
```
Comment (pmuetschard) [inline gapis/perfetto/android/trace.go]: just `cleanup`

File: gapis/perfetto/android/trace.go [hunk 2]
```diff
@@ -54,25 +90,28 @@ func Start(ctx context.Context, d adb.Device, a *android.ActivityAction, opts *s
...
+
 	log.I(ctx, "Unlocking device screen")
 	unlocked, err := d.UnlockScreen(ctx)
 	if err != nil {
 		log.W(ctx, "Failed to determine lock state: %s", err)
 	} else if !unlocked {
-		return nil, log.Err(ctx, nil, "Please unlock your device screen: GAPID can automatically unlock the screen only when no PIN/password/pattern is needed")
+		return nil, cleanupFunc, log.Err(ctx, nil, "Please unlock your device screen: GAPID can automatically unlock the screen only when no PIN/password/pattern is needed")
```
Comment (pmuetschard) [inline gapis/perfetto/android/trace.go]: `cleanup.Invoke()` here and all places you return an error

**Comment by lpy** [comment]
done, ptal

**Comment by lpy** [comment]
ping for review

File: gapis/perfetto/android/trace.go [hunk 3]
```diff
@@ -44,8 +50,38 @@ type Process struct {
...
+	}
+	if driverPackageOverride, err := d.SystemSetting(ctx, "global", prereleaseDriverOverrideSettingVariable); driverPackageOverride != "" && driverPackageOverride != "null" && err == nil {
+		driverPackageName = driverPackageOverride
+	}
+	if driverPackageName == "" {
+		return nil, nil
+	}
+	cleanup, err := android.SetupLayer(ctx, d, packageName, driverPackageName, renderStageVulkanLayerName, true)
+	if err != nil {
+		return nil, log.Err(ctx, err, "Failed to setup gpu.renderstages environment.")
```
Comment (pmuetschard) [inline gapis/perfetto/android/trace.go]: You missed the cleanup.Invoke() here!

**Pull request merged**